client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -136,7 +136,10 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 						break;
 				}
 			} );
-
+			// Jetpack returns an error if we try to save options that can't be changed.
+			if ( 'error_const' === fields.lang_id || 'error_cap' === fields.lang_id ) {
+				delete this.props.fields.lang_id;
```

Comment:
It's considered a bad practice to tamper with the `props` directly - they're supposed to be immutable and should not be changed directly by the component.

In addition, this will only work when clicking the save button. If we ever add an autosaving toggle or radio button to the same page that the language settings are displayed on, it will continue to fail, because we're only doing that for the submit handler, but not for any change handlers. 

So in order to have this fixed for all possible cases, we should do it a little later in the flow, for example in `WrappedSettingsForm.submitForm()`. 

Luckily, in `submitForm()` we have a `pick( fields, settingsFields.jetpack )`, where we define which fields to save. There we can just `omit` the `lang_id` from `settingsFields.jetpack` if it's `error_const` or `error_cap`.

Reply:
Good call, @tyxla. Thanks for the suggestion to put this in `submitForm`. I went ahead and did so in #22653.